src/System.Reflection.Metadata/src/System/Reflection/Metadata/Handles.cs

```diff
@@ -7,77 +7,179 @@
 
 namespace System.Reflection.Metadata
 {
+    /// <summary>
+    /// Represents any metadata entity (type reference/definition/specification, method definition, custom attribute, etc.) or value (string, blob, guid, user string).
+    /// </summary>
+    /// <remarks>
+    /// Use <see cref="Handle"/> to store multiple kinds of handles.
+    /// </remarks>
     public struct Handle : IEquatable<Handle>
     {
-        internal readonly uint value;
+        private readonly int _value;
 
-        internal Handle(uint value)
+        // bits:
+        //    7: IsVirtual
+        // 0..6: token type
+        private readonly byte _vType;
```

Comment:
Why can we not have just `Handle` and keep it 32-bits using the following:

1. Bits 0..1:
  * 11: Guid
  * 10: Blob
  * 01: String
  * 00: Other
2. Bit 2: IsVirtual
3. Remaining 29 bits:
  * For String/Blob/Guid, the 29-bit value
  * For everything else:
    * Bits 3..7: Token type
    * Bits 8..31: 24-bit value